From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-oa1-f41.google.com (mail-oa1-f41.google.com [209.85.160.41])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id 481B421CA0D
	for <linux-wireless@vger.kernel.org>; Thu, 19 Feb 2026 00:40:28 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.160.41
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1771461629; cv=none; b=KZmBZh3LnRHa8L8z6jr8BBStfY8qZ2oI8qbIv/1cTJbw7fDX+GIA9/LQk7U40R7i4JOefZwSGUgkVDIE6Ao44Jq6jrxCcZN5oRLDuc2haCMNuKTK8SSo4KvPsbaH7+uEkxJhFOies5sk0zJConFBZt6v8dOe0qZSdrJjly2+QpI=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1771461629; c=relaxed/simple;
	bh=gLbcxsvMfBs+N5v0NGbO8TPqMVB+Atm7jc9GjQ5hBTk=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version; b=VOX0SdgQU4sGdDhq1BAGHj8o9IgbBAbXkjAaH1F1Jov4FL4VoDlKrToe99FInxC11i1KYblxihUmLZNKWe99CnTRdCS+mDPS0HAHMTSVIOCVJMqdY1QE1w/0s6FEzEVzokSvuLAtre+9Fi/qwdbjS1cZEw9zyOFR3It4WXkbp34=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org; spf=pass smtp.mailfrom=gmail.com; arc=none smtp.client-ip=209.85.160.41
Authentication-Results: smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Received: by mail-oa1-f41.google.com with SMTP id 586e51a60fabf-40ed9a23aaeso128938fac.1
        for <linux-wireless@vger.kernel.org>; Wed, 18 Feb 2026 16:40:28 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1771461627; x=1772066427;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=kdvSYvr5NMYBqMkb8DQrSUZYlW2kwKMB4kHjYgxFfio=;
        b=itCbmxLHSTHR5ElBuGJU9l1j722pbmGWGsV6liZT11p+VFLLOX+SgBX0cdg+QvuUF3
         XtXJuVCsv6LPnUEzXEMsSsEyOoVtKaoVu/DJX8+AHw5R1RBSF6WJH/HS3Bswkubvc9La
         +XZ+aN2XykJYbeD7v2q1AbqOpU49NjUe2mOLPpFusKhpcac8oQtQq7wtcpglJL12WCr0
         l52qeqEYPOJ4JhfXXRZIGyScgY3dOnVYU1emDnzd9RB8buOhck/c99QfEGBo0Oj6fSmc
         AYE2G2Z1WsfHkBmDeyVKVdq8CZhVP0F2XqsQeKhDxdLcA1t0yn7Jmr1tBsUTVPxILpxi
         mpog==
X-Gm-Message-State: AOJu0YxBOIZfGt52LuBU4FzlD+Rd3p6VW6xlEuH21h/Jub6ICAZOyacN
	dBqSpE3NCP5NjwWGlEEKE40gs1X6U/rMD8hbyLaeaKa8PuTawsYeWJM6
X-Gm-Gg: AZuq6aJ1D3tIywiLSDKfIE+BPctvNsxDz5B0j4S3O5aDU1M79tKU9cDqtDnJ4aSEKlO
	WpNkKqY2IyuGLELdERZwOOGZVwwoq9VPiiz6RaYMTbN3ZwTF/Zf6LoLSlRj/M1vgEVwiFS8bV8j
	pAhE8jCXZ1iYQvFbGiZW1VMReZPTNHCtVHM1bUtuYH9V05Es6xA6D9qP+0zrOUgjYsbLQpDmzxa
	D6sqyb7x2h/s7GLiLueSWKUQr/03uTqqWDP1sEfhdkO9K9pFAlPkYXLvHgeEjusbb8j1yOp3JgE
	YwP0aZszioY4xnQQ5i1xGKCt+dHVuiRbv7fRVr8rkMy6i23cN9FQ0m9/o/K3vXrAD8BjgX4VRCa
	D8WZP88auLpnh7TCwAeMQHs+7fA5+4pd5ZmgLFxlKKbm8YHZ2qdDW8947NInthWds+Slnp+LR/C
	NqTAgKtaFnpeu3pPv29CsepmLjd9pd7mkjMo3Q69TSZ6YyMYPqSPygXsCUuA==
X-Received: by 2002:a05:6870:a0a6:b0:40e:e9f4:90f5 with SMTP id 586e51a60fabf-40f0d997a2bmr9443254fac.58.1771461627086;
        Wed, 18 Feb 2026 16:40:27 -0800 (PST)
Received: from sean-HP-EliteBook-830-G6.lan ([207.191.35.252])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-40eaee45d12sm23477659fac.1.2026.02.18.16.40.26
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Feb 2026 16:40:26 -0800 (PST)
From: sean.wang@kernel.org
To: nbd@nbd.name,
	lorenzo.bianconi@redhat.com
Cc: linux-wireless@vger.kernel.org,
	linux-mediatek@lists.infradead.org,
	Sean Wang <sean.wang@mediatek.com>,
	Xiong Huang <xiong.huang@mediatek.com>
Subject: [PATCH 09/11] wifi: mt76: mt792x: add MT7902 WFDMA prefetch configuration
Date: Wed, 18 Feb 2026 18:40:05 -0600
Message-ID: <20260219004007.19733-9-sean.wang@kernel.org>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260219004007.19733-1-sean.wang@kernel.org>
References: <20260219004007.19733-1-sean.wang@kernel.org>
Precedence: bulk
X-Mailing-List: linux-wireless@vger.kernel.org
List-Id: <linux-wireless.vger.kernel.org>
List-Subscribe: <mailto:linux-wireless+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-wireless+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

From: Sean Wang <sean.wang@mediatek.com>

Configure the RX/TX ring prefetch setting for MT7902 PCIe device.

This is a prerequisite patch before enabling MT7902 PCIe support.

Co-developed-by: Xiong Huang <xiong.huang@mediatek.com>
Signed-off-by: Xiong Huang <xiong.huang@mediatek.com>
Signed-off-by: Sean Wang <sean.wang@mediatek.com>
---
 drivers/net/wireless/mediatek/mt76/mt792x_dma.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/net/wireless/mediatek/mt76/mt792x_dma.c b/drivers/net/wireless/mediatek/mt76/mt792x_dma.c
index 34f07bd3097d..002aece857b2 100644
--- a/drivers/net/wireless/mediatek/mt76/mt792x_dma.c
+++ b/drivers/net/wireless/mediatek/mt76/mt792x_dma.c
@@ -103,6 +103,22 @@ static void mt792x_dma_prefetch(struct mt792x_dev *dev)
 		mt76_wr(dev, MT_WFDMA0_TX_RING3_EXT_CTRL, PREFETCH(0x0400, 0x10));
 		mt76_wr(dev, MT_WFDMA0_TX_RING15_EXT_CTRL, PREFETCH(0x0500, 0x4));
 		mt76_wr(dev, MT_WFDMA0_TX_RING16_EXT_CTRL, PREFETCH(0x0540, 0x4));
+	} else if (is_mt7902(&dev->mt76)) {
+		/* rx ring */
+		mt76_wr(dev, MT_WFDMA0_RX_RING0_EXT_CTRL, PREFETCH(0x0000, 0x4));
+		mt76_wr(dev, MT_WFDMA0_RX_RING1_EXT_CTRL, PREFETCH(0x0040, 0x4));
+		mt76_wr(dev, MT_WFDMA0_RX_RING2_EXT_CTRL, PREFETCH(0x0080, 0x4));
+		mt76_wr(dev, MT_WFDMA0_RX_RING3_EXT_CTRL, PREFETCH(0x00c0, 0x4));
+		/* tx ring */
+		mt76_wr(dev, MT_WFDMA0_TX_RING0_EXT_CTRL, PREFETCH(0x0100, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING1_EXT_CTRL, PREFETCH(0x0140, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING2_EXT_CTRL, PREFETCH(0x0180, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING3_EXT_CTRL, PREFETCH(0x01c0, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING4_EXT_CTRL, PREFETCH(0x0200, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING5_EXT_CTRL, PREFETCH(0x0240, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING6_EXT_CTRL, PREFETCH(0x0280, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING15_EXT_CTRL, PREFETCH(0x02c0, 0x4));
+		mt76_wr(dev, MT_WFDMA0_TX_RING16_EXT_CTRL, PREFETCH(0x0300, 0x4));
 	} else {
 		/* rx ring */
 		mt76_wr(dev, MT_WFDMA0_RX_RING0_EXT_CTRL, PREFETCH(0x0, 0x4));
-- 
2.43.0


